source/sigv4.c

```diff
@@ -1583,6 +1609,21 @@ static void generateCredentialScope( const SigV4Parameters_t * pSigV4Params,
     }
 
 /*-----------------------------------------------------------*/
+    static void storeHashedPayloadLocation( size_t headerIndex,
+                                            const char * pAmzSHA256Header,
+                                            CanonicalContext_t * canonicalRequest )
+    {
+        assert( canonicalRequest != NULL );
+
+        const uint8_t * pHeaderData = ( const uint8_t * ) canonicalRequest->pHeadersLoc[ headerIndex ].key.pData;
+        const uint8_t * pHeaderLiteral = ( const uint8_t * ) pAmzSHA256Header;
```

Comment:
Do we need an intermediate `pHeaderLiteral` variable here? We can just use `pAmzSHA256Header` and cast directly?

Reply:
Or better yet, just use the macro directly

Reply:
Yes we do. MISRA reasons

Reply:
This is done to avoid MISRA violation(doing the cast within the function call still triggers 21.16.)